extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/MicrosoftTransferExtension.java

```diff
@@ -48,51 +62,24 @@ public String getServiceId() {
 
   @Override
   public Exporter<?, ?> getExporter(String transferDataType) {
-    // TODO consider memoizing w/ Supplier but we will only use once and pass to transfer.
-    // This could allow us to refactor these params into supplier.
-    OkHttpClient client = new OkHttpClient.Builder().build();
-    ObjectMapper mapper = new ObjectMapper();
-    TransformerService transformerService = new TransformerServiceImpl();
-
-    if (transferDataType.equals(CONTACTS)) {
-      return new MicrosoftContactsExporter(BASE_GRAPH_URL, client, mapper, transformerService);
-    }
-    if (transferDataType.equals(CALENDAR)) {
-      return new MicrosoftCalendarExporter(BASE_GRAPH_URL, client, mapper, transformerService);
-    }
-
-    if (transferDataType.equals(PHOTOS)) {
-      return new MicrosoftPhotosExporter(BASE_GRAPH_URL, client, mapper, jobStore);
-    }
+    Preconditions.checkArgument(initialized);
```

Comment:
I believe checkState is more appropriate as this is not dependent on argument https://github.com/google/guava/wiki/PreconditionsExplained

```suggestion
    Preconditions.checkState(initialized);
```

Reply:
Done!